frontend/lib/parsing/parsing-queries.cpp

```diff
@@ -192,9 +192,32 @@ introspectParsedFiles(Context* context) {
   return toReturn;
 }
 
+static const BuilderResult&
+compilerGeneratedBuilderQuery(Context* context, UniqueString symbolPath) {
+  QUERY_BEGIN(compilerGeneratedBuilderQuery, context, symbolPath);
+
+  BuilderResult ret;
+
+  return QUERY_END(ret);
+}
+
 // parses whatever file exists that contains the passed ID and returns it
 const BuilderResult*
 parseFileContainingIdToBuilderResult(Context* context, ID id) {
+  {
+    UniqueString symbolPath = id.symbolPath();
+
+    while (!symbolPath.isEmpty()) {
+      auto tupleOfArgs = std::make_tuple(symbolPath);
+      auto got = context->hasCurrentResultForQuery(compilerGeneratedBuilderQuery, tupleOfArgs);
+      if (got) {
+        const BuilderResult& p = getCompilerGeneratedBuilder(context, symbolPath);
+        return &p;
+      }
+      symbolPath = ID::parentSymbolPath(context, symbolPath);
+    }
```

Comment:
This seems like a lot of work (for deeply nested IDs, it will keep trying to check the query database for more results, even if the ID is not compiler generated). Don't you already know if the ID is compiler-generated from its postorder ID being negative? It seems like you can skip this entire path if it _isn't_ negative.

Also, can compiler-generated modules like this be nested? Not currently, right? In that case, you may be able to do this way more efficiently by finding the outermost symbol and trying with it directly. I suspect you can use logic in `parentSymbolPath` as inspiration to do so; perhaps finding the first '.' (if any) and using that?